opensrp-chw-core/src/main/java/org/smartregister/chw/core/dao/StockUsageReportDao.java

```diff
@@ -0,0 +1,120 @@
+                " UNION ALL\n" +
+                " SELECT substr(mc.malaria_test_date, 7, 4) as \"Year\",\n" +
+                "        substr(mc.malaria_test_date, 4, 2) as \"Month\",\n" +
+                "\t\t'RDTs' as StockName,\n" +
+                "\t\tcount(1) as Usage\n" +
+                "\tfrom ec_malaria_confirmation mc\n" +
+                "where mc.malaria_results in('Negative','Positive')\n" +
+                "and mc.malaria_test_date is not NULL\n" +
+                "  group by mc.malaria_treat, substr(mc.malaria_test_date, 7, 4), substr(mc.malaria_test_date, 4, 2)\n" +
+                "order by Year DESC, Month DESC";
+
+        DataMap<StockUsage> dataMap = cursor -> {
+            StockUsage stockUsage = new StockUsage();
+            stockUsage.setYear(getCursorValue(cursor, "Year"));
+            stockUsage.setMonth(getCursorValue(cursor, "Month"));
+            stockUsage.setStockName(getCursorValue(cursor, "StockName"));
+            stockUsage.setStockUsage(getCursorValue(cursor, "Usage"));
+            return stockUsage;
+        };
+
+        List<StockUsage> res = readData(sql, dataMap);
+        if (res == null)
+            return new ArrayList<>();
+
+        return res;
+    }
+
+    /**
+     * Reads the min last update date and compares to today's date
+     * if last update date is not in same month as today return false
+     *
+     * @return
+     */
+    public static boolean lastInteractedWithinDay() {
+        String sql = "select count(stock_name) as count\n" +
+                "from stock_usage_report\n" +
+                "WHERE strftime('%Y',(updated_at)) == strftime('%Y',('NOW'))\n" +
+                "AND strftime('%m',(updated_at)) == strftime('%m',('NOW'))" +
+                " AND strftime('%d',(updated_at)) == strftime('%d',('NOW'))";
+        DataMap<String> dataMap = cursor -> getCursorValue(cursor, "count");
+
+        List<String> res = readData(sql, dataMap);
+        return (res != null && !(res.get(0).equals("0")));
+    }
+
+    public  String getStockUsageForMonth(String month, String stockName, String year) {
+        String sql = "select stock_usage from stock_usage_report " +
+                "WHERE month= '" + month + "' " +
+                "AND stock_name LIKE '%" + stockName + "%' " +
+                "AND year= '" + year + "'" +
+                "GROUP by stock_name";
+
+        DataMap<StockUsage> dataMap = cursor -> {
+            StockUsage stockUsage = new StockUsage();
+            stockUsage.setYear(year);
+            stockUsage.setMonth(month);
+            stockUsage.setStockName(stockName);
+            stockUsage.setStockUsage(getCursorValue(cursor, "stock_usage"));
+            return stockUsage;
+        };
+
+        List<StockUsage> res = readData(sql, dataMap);
+        if (res == null || res.size() == 0) {
+            return " ";
+        }
+        return res.get(0).getStockUsage();
```

Comment:
Why return a String instead of an int which is the count? An int might be easier to work with since it is a count of stock?